cmake/modules/extensions.cmake

```diff
@@ -1583,17 +1594,45 @@ function(zephyr_build_string outvar)
     )
   endif()
 
+  if(DEFINED BUILD_STR_BOARD AND BUILD_STR_SOC)
+    message(FATAL_ERROR
+      "zephyr_build_string(${ARGV0} <list> ...) given BOARD and SOC:"
+      " only one of these can be provided"
+    )
+  endif()
+
   if(DEFINED BUILD_STR_BOARD_REVISION AND NOT BUILD_STR_BOARD)
     message(FATAL_ERROR
       "zephyr_build_string(${ARGV0} <list> BOARD_REVISION ${BUILD_STR_BOARD_REVISION} ...)"
       " given without BOARD argument, please specify BOARD"
     )
   endif()
 
-  if(DEFINED BUILD_STR_BOARD_QUALIFIERS AND NOT BUILD_STR_BOARD)
+  if(DEFINED BUILD_STR_BOARD_QUALIFIERS AND NOT BUILD_STR_BOARD AND NOT BUILD_STR_SOC)
     message(FATAL_ERROR
       "zephyr_build_string(${ARGV0} <list> BOARD_QUALIFIERS ${BUILD_STR_BOARD_QUALIFIERS} ...)"
-      " given without BOARD argument, please specify BOARD"
+      " given without BOARD or SOC argument, please specify BOARD or SOC"
+    )
+  endif()
```

Comment:
All this code can be removed if we remove the need for `SOC` and instead allow `BOARD_QUALIFIERS` to be used without `BOARD`.

Reply:
Removed